src/Microsoft.ML.Core/Utilities/ResourceManagerUtils.cs

```diff
@@ -268,21 +278,28 @@ private Exception DownloadResource(IHostEnvironment env, IChannel ch, WebClient
                         size = 10000000;
 
                     long printFreq = (long)(size / 10.0);
-                    var buffer = new byte[4096];
+                    var buffer = new byte[blockSize];
                     long total = 0;
-                    int count;
+
+                    var task = s.ReadAsync(buffer, 0, blockSize);
+                    task.Wait(ct);
```

Comment:
ReadAsync allows you to pass a CancellationToken directly. You can choose to use that.